weather_server/weather_server/urls.py

```diff
@@ -19,16 +19,17 @@
 """
 from django.conf.urls import include, url
 from django.contrib import admin
-from show_weather.views import index, csv_observation_request, csv_stations
+from show_weather.views import index, csv_observation_request, csv_stations, png_observation_request
 
 data_patterns = [
-    url(r'^csv/'
-        r'([0-9]+)/'                                         # Station ID
-        r'([0-9]{4}-[0-9]{2}-[0-9]{2} [0-9]{2}:[0-9]{2})'    # Start date
+    url(r'^(?P<station>[0-9]+)/'                                         # Station ID
+        r'(?P<start>[0-9]{4}-[0-9]{2}-[0-9]{2} [0-9]{2}:[0-9]{2})'    # Start date
         r' - '
-        r'([0-9]{4}-[0-9]{2}-[0-9]{2} [0-9]{2}:[0-9]{2})$',  # End date
-        csv_observation_request),
-    url(r'^csv/stations/$', csv_stations)
+        r'(?P<end>[0-9]{4}-[0-9]{2}-[0-9]{2} [0-9]{2}:[0-9]{2})/', include([  # End date
+        url(r'csv$', csv_observation_request),
+        url(r'png$', png_observation_request),])),
```

Comment:
It would be nice to build this so that ``png`` and ``csv`` are at the beginning, not at the end. Not sure how to best implement that - off the top of my head I'm thinking something like putting those up front and then passing them to a ``date_parser`` or something, somehow passing it the harvested request type so that it knows what function to call. I prefer that because that would be more consistent with line 31, where the pattern *begins* with the request type. Before the merge we'll need to find a DRY way of expressing that, or modify the API for requestion a station list.